### PR TITLE
Add ability to use icons in text input.

### PR DESCRIPTION
**Pull request by mattias800**

### CODE REVIEW

**Comment by knegusen** [comment]
Screenshots!

File: src/components/ui/form/text-input/DefaultTextInput.tsx [hunk 1]
```diff
@@ -1,39 +1,169 @@
...
+  FocusHandlers;
+
+const DefaultTextInputComponent = ({
+  borderRadius,
+  borderStyle,
+  borderColor,
+  borderWidth,
+  theme,
+  focused,
+  iconLeft,
```
[the remark this answers is not in the document]
Comment (mattias800) [reply src/components/ui/form/text-input/DefaultTextInput.tsx]: I was thinking about it.

Pros for icons:
1. Easy and short, just iconLeft="ship", done.
2. User can't mess up the layout.
3. Better default values for icon size and color, so they are consistent everywhere.
4. Can be part of the theme.

Pros for ReactNode:
1. Can use any component.

Maybe support both?

**Comment by mattias800** [comment]
screen shot 2018-08-31 at 13 59 58
screen shot 2018-08-31 at 13 59 54
screen shot 2018-08-31 at 13 59 49
screen shot 2018-08-31 at 13 59 39

File: src/components/ui/form/text-input/DefaultTextInput.tsx [hunk 2]
```diff
@@ -1,39 +1,169 @@
...
 import { WithThemeProps } from '../../../util/enhancers/WithTheme';
+import { Border, BorderStyle } from '../../decorations';
+import { Icon } from '../../icon';
+import { Indent, Row } from '../../layout';
 import { SimpleTextInput, SimpleTextInputProps } from './SimpleTextInput';
-import { css } from 'emotion';
 
 export type __C12491142 = ComponentClass<{}>;
 
 // TODO Move to theme.
```
[the remark this answers is not in the document]
Comment (mattias800) [reply src/components/ui/form/text-input/DefaultTextInput.tsx]: Heeeeelp meeeee! 😭 😭 😭

File: src/components/ui/form/text-input/SimpleTextInput.tsx [hunk 3]
```diff
@@ -198,14 +220,17 @@ class SimpleTextInputComponent extends React.Component<
         }}
         className={className}
         type="text"
-        ref={input => (this.textInput = input)}
+        ref={(input: any) => (this.textInput = input)}
```
Comment (knegusen) [inline src/components/ui/form/text-input/SimpleTextInput.tsx]: Is it ok to use `any` as type?

**Pull request merged**